src/endpoints/resources/impl.cpp

```diff
@@ -498,7 +499,45 @@ namespace
                 json::object_t info;
                 bool exists = false;
 
-                if (const auto resc = adm::client::resource_info(conn, name_iter->second); resc) {
+                const auto& config = irods::http::globals::configuration();
+
+                if (config.at(json::json_pointer{"/irods_client/enable_4_2_compatibility"}).get<bool>()) {
+                    const auto gql = fmt::format("select RESC_ID, RESC_TYPE_NAME, RESC_ZONE_NAME, "
+                                                 "RESC_LOC, RESC_VAULT_PATH, RESC_STATUS, "
+                                                 "RESC_CONTEXT, RESC_COMMENT, RESC_INFO, "
+                                                 "RESC_FREE_SPACE, RESC_FREE_SPACE_TIME, "
+                                                 "RESC_PARENT, RESC_CREATE_TIME, RESC_MODIFY_TIME "
+                                                 "where RESC_NAME = '{}'",
+                                                 name_iter->second);
+
+                    for (auto&& row : irods::experimental::query_builder{}.build<RcComm>(conn, gql)) {
+                        exists = true;
+
+                        info = { // TODO Can't use += yet
```

Comment:
we have an issue for this?

Reply:
We don't. That TODO is from the irods/irods code.

Will remove since it's not required.

Reply:
Removed.